src/output_gstreamer.h

```diff
@@ -25,6 +25,73 @@
 #ifndef _OUTPUT_GSTREAMER_H
 #define _OUTPUT_GSTREAMER_H
 
-extern struct output_module gstreamer_output;
+#include <gst/gst.h>
 
-#endif /*  _OUTPUT_GSTREAMER_H */
+#include "output_module.h"
+
+class GstreamerOutput : public OutputModule,
+                        public OutputModuleFactory<GstreamerOutput> {
+ public:
+  struct Options : public OutputModule::Options {
+    // Let GstreamerOutput access protected constructor
+    friend class GstreamerOutput;
+
+    char* audio_sink = nullptr;
```

Comment:
do these need to be writeable `char*` ? These look like they want to be `const char*`.